backend/routes/auth.js

```diff
@@ -0,0 +1,39 @@
+const express = require("express");
+app.use(express.json());
+const app = express();
+app.use(express.json());
+router = express.Router();
+const argon2 = require("argon2");
+
+router.post("/signup", async (req, res) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    return res
+      .status(400)
+      .json({ error: "Username and password are required." });
+  }
+
+  if (password.length < 8) {
+    return res
+      .status(400)
+      .json({ error: "Password must be at least 8 characters long." });
```

Comment:
Use the const here, in case you decide to change the logic and make it for example 12, so that you'll have only one place to update.